src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java

```diff
@@ -55,267 +56,121 @@
-                        ((SynchronizeFolderOperation) operation).getFolderPath()).getName();
-                if (result.getCode() == ResultCode.FILE_NOT_FOUND) {
-                    message = String.format(
-                            res.getString(R.string.sync_current_folder_was_removed),
-                            folderPathName
-                    );
-                }
-            }
-
-        } else if (operation instanceof CopyFileOperation) {
-            if (result.getCode() == ResultCode.FILE_NOT_FOUND) {
-                message = res.getString(R.string.copy_file_not_found);
-
-            } else if (result.getCode() == ResultCode.INVALID_COPY_INTO_DESCENDANT) {
-                message = res.getString(R.string.copy_file_invalid_into_descendent);
-
-            } else if (result.getCode() == ResultCode.INVALID_OVERWRITE) {
-                message = res.getString(R.string.copy_file_invalid_overwrite);
+    public static String getErrorCauseMessage(final RemoteOperationResult result,
+                                           final RemoteOperation operation,
+                                           final Resources resources) {
+
+        final RemoteOperationResult rlt = result;
+        final RemoteOperation o = operation;
+        Formater f = new Formater(resources);
+        if(rlt.isSuccess()) {
+            if(o instanceof UploadFileOperation) return f.f(R.string.uploader_upload_succeeded_content_single, ((UploadFileOperation) o).getFileName());
+            if(o instanceof DownloadFileOperation) return f.f(R.string.downloader_download_succeeded_content);
+            if(o instanceof RemoveFileOperation) return f.f(R.string.remove_success_msg);
+        }
 
-            } else if (result.getCode() == ResultCode.FORBIDDEN) {
-                message = String.format(res.getString(R.string.forbidden_permissions),
-                        res.getString(R.string.forbidden_permissions_copy));
+        if(o instanceof SynchronizeFileOperation &&
+                !((SynchronizeFileOperation) o).transferWasRequested()) {
+            return f.f(R.string.sync_file_nothing_to_do_msg);
+        }
 
-            }
+        if((o instanceof CreateShareViaLinkOperation
+                || o instanceof RemoveShareOperation
+                || o instanceof UpdateShareViaLinkOperation
+                || o instanceof UpdateSharePermissionsOperation)
+                && rlt.getData() != null
+                && rlt.getData().size() > 0) {
+            return rlt.getData().get(0).toString();
         }
 
-        return message;
+        switch (rlt.getCode()) {
+            case LOCAL_STORAGE_FULL:
+            case LOCAL_STORAGE_NOT_COPIED: return f.f(R.string.error__upload__local_file_not_copied, ((UploadFileOperation) o).getFileName(),R.string.app_name);
+            case FORBIDDEN:
+                if(o instanceof UploadFileOperation) f.f(R.string.forbidden_permissions, R.string.uploader_upload_forbidden_permissions);
```

Comment:
Why are you using return in some cases and not in others?

Reply:
Because I forgot them ... it's a little downside of this why of coding. Thank you for noticing.